pkg/kcp/provider/azure/vpcpeering/peeringRemoteDelete.go

```diff
@@ -13,21 +15,27 @@ func peeringRemoteDelete(ctx context.Context, st composed.State) (error, context
 		return nil, nil
 	}
 
-	if len(state.ObjAsVpcPeering().Status.RemoteId) == 0 {
-		logger.Info("Remote VpcPeering deleted before Azure peering is created")
+	if state.remotePeering == nil {
+		logger.Info("Skipping remote peering deletion")
```

Comment:
Perhaps a better info message would be something along lines "remote peering not loaded, continuing".
Current message might make it seem like we know there is a peering, but we are choosing not to delete it.